src/unlimited.h

```diff
@@ -196,6 +201,7 @@ extern CTweak<uint64_t> blockSigopsPerMb;
 extern CTweak<uint64_t> coinbaseReserve;
 extern CTweak<uint64_t> blockMiningSigopsPerMb;
 
+extern std::list<CStatBase*> mallocedStats;
```

Comment:
What about just using a `stack` or `vector` for this? `push_back` `pop_back` exist for `vector` so you can still preserve cleanup order.